src/Base/PyTools.c

```diff
@@ -264,7 +270,11 @@ void PP_Fetch_Error_Text()
         PyObject* value = PyDict_GetItemString(errdata,"swhat");
         
         if (value!=NULL) {
-            strncpy(PP_last_error_info, PyString_AsString(value), MAX);
+#if PY_MAJOR_VERSION < 3
+            strncpy(PP_last_error_info, PyString_AsString(pystring), MAX);

```

Comment:
I just realized the regression of this change. The argument is not pystring but value. Since pystring is NULL it causes a segfault.